### DISCUSSION TITLE
How to create own gio.ListModeller for use in ListView

### DISCUSSION THREAD

**Discussion by abenz1267** [diamondburned/gotk4#136] [Discussions]
Hi,

i'm currently trying to implement my own ListModeller so I can use Go struct types as value for a ListView. Is there and "easy" way I missed, or do I have to manually implement `gio.ListModeller`, which itself extends `glib.Objector`. 

I "just" want to do `gtk.NewSingleSelection([]MyStructType{...}`.

Anyone done anything here?

Regards

**Comment by diamondburned** [comment]
You could use [gioutil.ListModel](https://pkg.go.dev/github.com/diamondburned/gotk4/pkg/core/gioutil#ListModel).

**Comment by abenz1267** [comment]
Thanks a lot !

Implemented it this way, technically it works, but i'm getting 

```
ERROR gotk4_gbox_list_get_item: assertion 'index < objects_get_size(&self->items)' failed glib_old_log_api=1 priority=4
```

Every time the list changes.

**Reply by diamondburned** [reply]
What's your code?

**Reply by abenz1267** [reply]
```
	items := gioutil.NewListModel[modules.Entry]()
	gtk.NewSingleSelection(items.ListModel)
```
